pkg/common/peertracker/credentials_test.go

```diff
@@ -0,0 +1,32 @@
+package peertracker
+
+import (
+	"bytes"
+	"errors"
+	"io/ioutil"
+	"net"
+	"testing"
+
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/require"
+)
+
+func TestServerHandshakeCatchesAcceptError(t *testing.T) {
+	var out bytes.Buffer
+	logger := logrus.New()
+	logger.Level = logrus.WarnLevel
+	logger.Out = &out
+	wrappedConn := &Conn{
+		Conn: &net.UnixConn{},
+		Err:  errors.New("bad accept"),
+	}
+	creds := NewCredentials(logger)
+
+	_, _, err := creds.ServerHandshake(wrappedConn)
+	require.EqualError(t, err, "bad accept")
+
+	msg, err := ioutil.ReadAll(&out)
+	require.NoError(t, err)
+	require.Contains(t, string(msg), `msg="Connection failed during accept."`)
```

Comment:
nit: these assertions could be a little more tight if we used a logrus test hook and inspected the log entries instead of inspecting the string output.

Reply:
good call, i was being lazy :D, fixed